### DISCUSSION TITLE
Shadow DOM?

### DISCUSSION THREAD

**Discussion by keichan34** [maplibre/maplibre-gl-js#1304] [Ideas]
Recently I've been embedding MapLibre maps in a lot of different websites, and some sites have CSS rules that are too broad and leak in to the MapLibre map and/or controls (attribution being set as `display: none` when it should have been block, for example). Increasing the specificity of the rules seem to help with some sites, but sometimes there are IDs in the CSS and I'll have to use something like `!important` or add my own IDs.

In more recent sites, I've decided to just wrap the whole map in a Shadow DOM to solve the problem of outside CSS interfering with the map, and that's been working well.

I was wondering if a feature like this would be appropriate to put in to the library itself?

**Comment by HarelM** [comment]
Leaking CSS seems to be related to how you build the site around maplibre.
The css in maplibre is fairly specific I believe.
From my point of view, I don't see something that needs to be incorporated into maplibre.
I might be wrong though...
